Userland/Libraries/LibCore/System.cpp

```diff
@@ -528,9 +528,9 @@ ErrorOr<int> anon_create([[maybe_unused]] size_t size, [[maybe_unused]] int opti
         return Error::from_errno(saved_errno);
     }
 #elif defined(AK_OS_BSD_GENERIC) || defined(AK_OS_EMSCRIPTEN) || defined(AK_OS_HAIKU)
-    struct timespec time;
-    clock_gettime(CLOCK_REALTIME, &time);
-    auto name = ByteString::formatted("/shm-{}{}", (unsigned long)time.tv_sec, (unsigned long)time.tv_nsec);
+    static size_t shared_memory_id = 0;
```

Comment:
If we were to be extremely paranoid - should this be thread safe as well? It doesn't seem something that is likely in practice, but it's low level enough down in the stack that made me think about it

Reply:
I had thought about making this ID `thread_local`, but then we'd want to add the current thread ID to the shm name, which is a very grossly non-portable concept. In this platform branch, we'd have to support macOS, freebsd, openbsd, netbsd, haiku, etc., and they all have different ways of getting the thread ID - so I figured we'd cross that bridge if we ever get there 😅 At least if there is ever an issue, we'll get a traceable crash.